_includes/exhibits.js

```diff
@@ -0,0 +1,1929 @@
+    design.layout = _layout;
+    this.design = design;
+    this.changed = true;
+  },
+
+  get images() {
+    return this.design.images || [];
+  },
+  set images(_images) {
+    var design = this.design;
+    design.images = _images;
+    this.design = design;
+    this.changed = true;
+  },
+
+  get grid() {
+    return unpackGrid(this.layout, this.images, 'Grid');
+  },
+
+  get target() {
+    return unpackGrid(this.layout, this.images, 'Target');
+  },
+
+  /*
+   * Derived State
+   */
+
+  get story() {
+    return this.stories[this.s];
+  },
+  set story(story) {
+    const stories = this.stories;
+    stories[this.s] = story;
+    this.stories = stories;
+  },
+
+  get group() {
+    return this.cgs[this.g];
+  },
+
+  get colors() {
+    return this.group.Colors;
+  },
+
+  get channels() {
+    return this.group.Channels;
+  },
+
+  get waypoints() {
+    return this.story.Waypoints;
+  },
+  set waypoints(waypoints) {
+    const story = this.story;
+    story.Waypoints = waypoints;
+    this.story = story;
+  },
+
+  get waypoint() {
+    if (this.editing) {
+      return this.bufferWaypoint;
+    }
+    return this.waypoints[this.w];
+  },
+  set waypoint(waypoint) {
+    if (this.editing) {
+      this.bufferWaypoint = waypoint;
+    }
+    else {
+      const waypoints = this.waypoints;
+      waypoints[this.w] = waypoint;
+      this.waypoints = waypoints;
+    }
+  },
+
+  get viewport() {
+    const v = this.v;
+    return {
+      scale: v[0],
+      pan: new OpenSeadragon.Point(v[1], v[2])
+    };
+  },
+
+  get overlay() {
+    const o = this.o;
+    return {
+      x: o[0],
+      y: o[1],
+      width: o[2],
+      height: o[3]
+    };
+  },
+
+  /*
+   * State manaagement
+   */
+
+  matchQuery: function(hash, hashKeys) {
+    const keys = Object.keys(hash);
+    if (keys.length != hashKeys.length) {
+      return false;
+    }
+    return hashKeys.reduce(function(accept, key) {
+      return accept && hash[key] !== undefined;
+    }, true);
+  },
+
+  newExhibit: function() {
+    const exhibit = this.exhibit;
+    const cgs = deepCopy(exhibit.Groups || []);
+    const stories = deepCopy(exhibit.Stories || []);
+    this.design = {
+      chans: deepCopy(exhibit.Channels || []),
+      layout: deepCopy(exhibit.Layout || {}),
+      images: deepCopy(exhibit.Images || []),
+      stories: stories,
+      cgs: cgs
+    };
+  },
+  newTag: function() {
+    const exhibit = this.exhibit;
+    const stories = deepCopy(exhibit.Stories);
+    const group = this.group;
+    const o = this.o;
+    const v = this.v;
+    const d = this.d;
+    this.stories = [{
+      Description: '',
+      Name: 'Tag',
+      Waypoints: [{
+        Zoom: v[0],
+        Pan: v.slice(1),
+        Group: group.Name,
+        Description: decode(d),
+        Name: 'Tag',
+        Overlay: {
+          x: o[0],
+          y: o[1],
+          width: o[2],
+          height: o[3],
+        },
+      }]
+    }].concat(stories);
```

Comment:
This uses the shareable description and coordinates in the url to make a new story that gets added to the front of the list of stories.